apps/src/templates/instructions/Instructions.jsx

```diff
@@ -70,26 +89,88 @@ class Instructions extends React.Component {
     }
   }
 
+  componentDidMount() {
+    let largestHeight = 0;
+
+    // Find the tallest of our dynamic instructions.
+    if (this.props.dynamicInstructions) {
+      Object.keys(this.props.dynamicInstructions).forEach(key => {
+        const height = $(
+          ReactDOM.findDOMNode(this.dynamicInstructionsRefs[key])
+        ).outerHeight(true);
+        if (height > largestHeight) {
+          largestHeight = height;
+        }
+      });
+    }
+
+    // Resize the parent div to be the height of the largest dynamic instruction.
+    if (this.props.setInstructionsRenderedHeight) {
+      this.props.setInstructionsRenderedHeight(
+        largestHeight + HEADER_HEIGHT + RESIZER_HEIGHT + 20
+      );
+    }
+  }
```

Comment:
maybe this is a case no one cares about and not one I'd expect you to address in this change - but what happens if you switch languages mid-level and the translation is too long for the tallest set height.

Reply:
We reload the entire page upon a language change, so we should get such adjustments for free.

That said, this touches upon a complication I came across earlier when the strings were not known until display time; we weren't able to do this measurement.  So, one limitation of this current implementation is that all dynamic instruction strings must be set at initial render time.